api/v1alpha1/searchoperator_types.go

```diff
@@ -44,6 +44,7 @@ type SearchOperatorSpec struct {
 type SearchOperatorStatus struct {
 	// Reflects the current status of the RedisGraph pod using a Persistence mode (PVC/EmptyDir/Degraded)
 	PersistenceStatus string `json:"persistence"`
+	PreviousState     *bool  `json:"previousstate,omitempty"`
```

Comment:
The name `previousstate` can be confusing. Can we call it `DeployRedisgraph` ?

Reply:
Do we want `omitempty` here?

```
The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.
```
https://golang.org/pkg/encoding/json/

From that I understand that if the value is `false` then the CR won't have this field.

Reply:
Since the type is *bool, it should be omitted only if it is nil. If omitempty is not added, it will be marked as required. Not sure if we want to make it a required field - will it break existing operator CRs?